src/App/Document.cpp

```diff
@@ -3817,7 +3822,12 @@ DocumentObject * Document::getActiveObject() const
     return d->activeObject;
 }
 
-DocumentObject * Document::getObject(const char *Name) const
+DocumentObject * Document::getObject(const char* Name) const
+{
+    return getObject(std::string(Name));

```

Comment:
Doing it this way is bad because every time when calling the function a tmp. std::string will be created. It makes much more sense to have the overloaded method:
```
DocumentObject * Document::getObject(const std::string& Name) const
{
    return getObject(Name.c_str());
}
```